automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingProjectOwnerPermissionSpec.scala

```diff
@@ -0,0 +1,270 @@
+            AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+          }
+        }
+      }
+    }
+
+    "+ project owner, + group member, create workspace, - project owner" in {
+      val owner = UserPool.chooseProjectOwner
+      val creator = owner
+      val user = UserPool.chooseStudent
+      implicit val token: AuthToken = creator.makeAuthToken()
+
+      withCleanBillingProject(creator, ownerEmails = List(user.email)) { projectName =>
+        withGroup("group", List(user.email)) { groupName =>
+          withWorkspace(projectName, "GroupApiSpec_workspace", Set(groupName)) { workspaceName =>
+            AuthDomainMatcher.checkVisibleAndAccessible(projectName, workspaceName, List(groupName))(user.makeAuthToken())
+
+            Rawls.billing.removeUserFromBillingProject(projectName, user.email, BillingProjectRole.Owner)
+            AuthDomainMatcher.checkNotVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+          }
+        }
+      }
+    }
+
+    "+ project owner, create workspace, + group member, - group member" in {
+      val owner = UserPool.chooseProjectOwner
+      val creator = UserPool.chooseStudent
+      val user = owner
+      implicit val token: AuthToken = creator.makeAuthToken()
+
+      withCleanBillingProject(owner, List(creator.email)) { projectName =>
+        withGroup("group") { groupName =>
+          withWorkspace(projectName, "GroupApiSpec_workspace", Set(groupName)) { workspaceName =>
+
+            AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+
+            Sam.user.addUserToPolicy(groupName, GroupRole.Member.toString, user.email)
+            register cleanUp Try(Sam.user.removeUserFromPolicy(groupName, GroupRole.Member.toString, user.email)).recover {
+              case _: RestException =>
+            }
+            AuthDomainMatcher.checkVisibleAndAccessible(projectName, workspaceName, List(groupName))(user.makeAuthToken())
+
+            Sam.user.removeUserFromPolicy(groupName, GroupRole.Member.toString, user.email)
+            AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+          }
+        }
+      }
+    }
+
+    "+ project owner, create workspace, + group member, - project owner" in {
+      val owner = UserPool.chooseProjectOwner
+      val creator = owner
+      val user = UserPool.chooseStudent
+      implicit val token: AuthToken = creator.makeAuthToken()
+
+      withCleanBillingProject(creator, ownerEmails = List(user.email)) { projectName =>
+        withGroup("group") { groupName =>
+          withWorkspace(projectName, "GroupApiSpec_workspace", Set(groupName)) { workspaceName =>
+
+            AuthDomainMatcher.checkVisibleNotAccessible(projectName, workspaceName)(user.makeAuthToken())
+
+            Sam.user.addUserToPolicy(groupName, GroupRole.Member.toString, user.email)
```

Comment:
Sam doesn't need to be called for these tests. All the add / remove users from groups should be down through the rawls api

Reply:
I don't know what Rawls api to use for add/remove users from groups. Can you tell me what they are and an example?

Reply:
hmm, looking more into it, seems orch is just calling sam directly for this rather than going through rawls. This is fine then!